temurin-{,jre-}bin-23: drop, open{jdk,jfx}23: drop

galderz opened the pull request:
With the release of Temurin 24, version 23 is no longer supported so users should not be using that any more since that is not an LTS (21 and 17 are). So, removing Temurin 23 to avoid confusion.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

galderz committed:
temurin-{,jre-}bin-23: drop

Co-Authored-By: Leona Maroni <dev@leona.is>

galderz commented:
Looking into the CI failures. I'll amend commit message too.

galderz commented:
Fixed commit messages. Added aliases as done in https://github.com/NixOS/nixpkgs/pull/344544/commits/d19c7e8b164b2803ab5ad6efdb5fb45973c7d4a2 when Temurin 22 was dropped.

FliegendeWurst commented:
I think you need to drop openjdk 23 first, it depends on temurin for building.

galderz committed:
openjdk23{,_headless}: drop

Co-Authored-By: Leona Maroni <dev@leona.is>

galderz committed:
treewide: migrate openjdk-23 -> openjdk-25

Co-Authored-By: Leona Maroni <dev@leona.is>

galderz commented:
Good point, I've removed openjdk 23 now and adjusted dependencies.

FliegendeWurst commented on pkgs/top-level/aliases.nix:
```diff
@@ -1791,6 +1791,8 @@ mapAliases {
   temurin-jre-bin-22 = throw "Temurin 22 has been removed as it has reached its end of life"; # Added 2024-09-24
   temurin-bin-22 = throw "Temurin 22 has been removed as it has reached its end of life"; # Added 2024-09-24
+  temurin-jre-bin-23 = throw "Temurin 23 has been removed as it has reached its end of life"; # Added 2025-05-01
+  temurin-bin-23 = throw "Temurin 23 has been removed as it has reached its end of life"; # Added 2025-05-01
```
Probably we want these for jdk23 too

galderz replied:
Added them

FliegendeWurst commented:
`openjfx23` needs to be dropped in favor of `openjfx24`. In fact this should have been done as part of the update to JDK 24 already.

leona-ya committed:
treewide: migrate openjfx23 -> openjfx25

While pdfsam-basic maintainer said that they require OpenJFX 23,
OpenJFX 25 works fine too. As version 23 is EOL, migrate to 25.

Co-Authored-By: Galder Zamarreño <galder@zamarreno.com>

galderz committed:
openjfx23: drop

FliegendeWurst commented on pkgs/top-level/all-packages.nix:
```diff
@@ -13024,7 +13015,7 @@ with pkgs;
   jabref = callPackage ../applications/office/jabref {
     jdk = jdk21.override {
```
Combining JDK 21 and OpenJFX 24 seems a bit sketchy. But it previously combined JDK 21 and OpenJFX 23, so this upgrade can't make it worse.

FliegendeWurst reviewed (approved):
LGTM. Didn't check any of the individual package changes.

galderz commented:
I've fixed the conflicts

eclairevoyant commented:
@galderz Looks like there's anither rebase needed

leona-ya commented:
I rebased the PR (mostly conflicts as this PR tried to migrate to temurin 24, but this has already been removed as it's EOL). I also updated the commit messages to fit the current situation and also generally cleaned them up.

galderz commented:
Thanks @leona-ya

leona-ya commented:
this lies around for so long, I tested with nixpkgs-review that it works and this is also a 25.11 release blocker. Let's go forward and merge this.
